src/mame/video/atarimo.cpp

```diff
@@ -438,7 +439,7 @@ void atari_motion_objects_device::render_object(bitmap_ind16 &bitmap, const rect
 	// extract data from the various words
 	int code = m_codelookup[rawcode];
 	int color = m_colorlookup[m_colormask.extract(entry)];
-	int xpos = m_xposmask.extract(entry);
+	int xpos = m_xposmask.extract(entry) - m_xoffset;
```

Comment:
Please reverse the sign of this offset so it can be added like the ones in the VAD are.

Reply:
I did it this way because it keeps the shift directions the same, if you add to a scroll value you shift left, if you subtract from an xposition you also shift left.

Reply:
Oh, so they have opposite directions.  In that case, maybe it would make sense to reverse all of them.  Generally we think of a positive displacement as moving to the right in computer graphics.

Reply:
afaik that would be a project-wide change?

Reply:
No, I just meant for these layer offset setters.  How do other devices with layer offset setters behave?  Examples like Namco C45 road X offset, Toaplan SCU offsets, X1-012 offsets, and so on.  I would have expected positive offsets to move things down and to the right.

Reply:
anyway i've made the requested change for now

Reply:
If all the existing video devices with configurable layer offsets treat a positive offset as moving to the left, I’m fine with these ones working that way, but it’s not what I would have expected.

Reply:
a positive value on m_tilemap->set_scrollx using the tilemap system moves the tilemap to the left.  Every MAME driver using tilemaps is built around that.

Reply:
I'm not sure there is a right/wrong answer here, the way I initially did it was because it made it easier when comparing with the videos.  I've changed it to the way you requested, which may be more logical in some cases.

Reply:
But one can equally say “a more positive value of a sprite coordinate moves the sprite to the right.  At a glance:
* video/decospr.cpp adds m_x_offset and m_y_offset to the X and Y coordinates when the screen isn’t flipped, so positive offsets move sprites down and to the right
* video/kan_pand.cpp adds m_xoffset and m_yoffset to the X and Y coordinates, so positive offsets move sprites down and to the right
* video/namco_c123tmap.cpp adds m_xoffs and m_yoffs to the tilemap scroll values, so positive offsets move the layer up and to the left
* video/tecmo_spr.cpp adds m_yoffset to the Y coordinates, so a positive offset moves sprites down
* video/vsystem_spr.cpp adds m_xoffs and m_yoffs to the X and Y coordinates, so positive offsets move sprites down and to the right
* video/x1_012.cpp does the opposite, subtracting the m_xoffsets[...] from the X coordinate, so positive offsets move sprites to the left

There isn’t even complete consensus between the existing devices on which direction positive offsets should move a layer, but sprite devices tend to treat positive offsets as meaning down and to the right.

Reply:
well it should match the most common behaviors with the current code at least